spark-java/src/main/java/com/baeldung/sparkjava/UserServiceMapImpl.java

```diff
@@ -0,0 +1,45 @@
+package com.baeldung.sparkjava;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+
+public class UserServiceMapImpl  implements UserService{
+    private HashMap<String, User> userMap;
+    
+    public UserServiceMapImpl() {
+        userMap = new HashMap<>();
+    }
+
+    public void addUser (User user) {
+        userMap.put(user.getId(), user);
+    }
+    
+    public Collection<User> getUsers () {
+        return  userMap.values();
+    }
+    
+    public User getUser (String id) {
+        return userMap.get(id);
+    }
+    
+    public User editUser (String id, HashMap userArg) throws Exception{
```

Comment:
I wanted to do the same and hence wanted to go for reflection solution, if you remember I asked you earlier. My problem was what happens to those fields which are not "set" and passed in PUT request. 

Hence the Map solutions. Please suggest if current approach is not doable.